api/solarperformanceinsight_api/compute.py

```diff
@@ -444,3 +507,293 @@ def compare_expected_and_actual(job: models.StoredJob, si: storage.StorageInterf
+        pac_adj = pac_ref_adj.clip(upper=pac0)  # type: ignore
+        pac_adj.index.name = "time"
+        total_ref_pac += pac_adj["performance"]
+        results_list.append(
+            DBResult(
+                schema_path=f"/inverters/{i}",
+                type="weather adjusted performance",
+                data=pac_adj,
+            )
+        )
+    return results_list, total_ref_pac
+
+
+def compare_predicted_and_actual(job: models.StoredJob, si: storage.StorageInterface):
+    results_list, total_ref_pac = _calculate_weather_adjusted_predicted_performance(
+        job, si
+    )
+    actual_monthly_energy, months = _get_actual_monthly_energy(job, si)
+    ref_energy = total_ref_pac.resample("1h").mean()  # type: ignore
+    ref_monthly_energy = (
+        ref_energy.groupby(ref_energy.index.month).sum().reindex(months)
+    )
+    diff = actual_monthly_energy - ref_monthly_energy
+    ratio = actual_monthly_energy / ref_monthly_energy
+    comparison_summary = pd.DataFrame(
+        {
+            "actual_energy": actual_monthly_energy,
+            "weather_adjusted_energy": ref_monthly_energy,
+            "difference": diff,
+            "ratio": ratio,
+        }
+    )
+    month_name_index = pd.Index([calendar.month_name[i] for i in months], name="month")
+    comparison_summary.index = month_name_index
+    results_list.append(
+        DBResult(
+            schema_path="/",
+            type="actual vs weather adjusted reference",
+            data=comparison_summary,
+        )
+    )
+    save_results_to_db(job.object_id, results_list, si)
+
+
+def compare_monthly_predicted_and_actual(
+    job: models.StoredJob, si: storage.StorageInterface
+):
+    data_ids_by_type = {do.definition.type: do.object_id for do in job.data_objects}
+    ref_weather = _get_data(
+        job.object_id,
+        data_ids_by_type[models.JobDataTypeEnum.monthly_original_weather],
+        si,
+    )
+    actual_weather = _get_data(
+        job.object_id,
+        data_ids_by_type[models.JobDataTypeEnum.monthly_actual_weather],
+        si,
+    )
+    ref_perf = _get_data(
+        job.object_id,
+        data_ids_by_type[models.JobDataTypeEnum.monthly_original_performance],
+        si,
+    )["total_energy"]
+    actual_perf = _get_data(
+        job.object_id,
+        data_ids_by_type[models.JobDataTypeEnum.monthly_actual_performance],
+        si,
+    )["total_energy"]
+
+    inverters = job.definition.system_definition.inverters
+    pac0 = sum([inv.inverter_parameters._pac0 for inv in inverters])
+    G_0 = 1000
+
+    gammas = [arr._gamma for inv in inverters for arr in inv.arrays]
+    if any([g is None for g in gammas]):
+        raise TypeError(
+            "Currently unable to compare predicted and actual performance for "
+            "PVsyst specified arrays."
+        )
+
+    avg_gamma = sum(
+        [
+            sum([arr._gamma for arr in inv.arrays]) / len(inv.arrays)  # type: ignore
```

Comment:
Does this work?

```python
from statistics import mean
avg_gamma = mean([
    mean([arr._gamma for arr in inv.arrays])
    for inv in inverters
])
```

e.g. 

```python
nested_list = [[1, 2, 3], [1, 2], [2, 3]]
mean([
    mean([a for a in b])
    for b in nested_list
])
2.0
```

Reply:
wow, did not know the statistics module existed...